lib/string_reverse.rb

```diff
@@ -1,4 +1,21 @@
 # A method to reverse a string in place.
 def string_reverse(my_string)
-  my_string << "not implemented"
+
+  return if my_string == nil || my_string.length == 0
+
+
+  i = 0
+  j = my_string.length - 1
+  temp = 0
+
+  while i < j
+    temp = my_string[i]
+    my_string[i] = my_string[j]
+    my_string[j] = temp
+
+    i += 1
+    j -= 1
+  end
+
+    return my_string

```

Comment:
Minor: since this method reverses the string in-place, you don't need to return the string. Line 20 could simply be `return`